src/XMakeBuildEngine/UnitTestsPublicOM/Definition/ProjectItem_Tests.cs

```diff
@@ -442,42 +441,98 @@ public void ExcludeVectorWithIncludeVector()
-        /// </summary>
-        [Fact]
-        public void Wildcards()
-        {
-            string directory = null;
-            string file1 = null;
-            string file2 = null;
-            string file3 = null;
-
+        private const string ExcludeWithWildCards = @"
+                    <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003' >
+                        <ItemGroup>
+                            <i Include='{0}' Exclude='{1}'/>
+                        </ItemGroup>
+                    </Project>
+                ";
+        [Theory]
+        [InlineData(
+            ExcludeWithWildCards,
+            "a.*",
+            "*.1",
+            new[] { "a.1", "a.2", "a.1" },
+            new[] { "a.2" })]
+        [InlineData(
+            ExcludeWithWildCards,
+            @"**\*.cs",
+            @"a\**",
+            new[] { "1.cs", @"a\2.cs", @"a\b\3.cs", @"a\b\c\4.cs" },
+            new[] { "1.cs" })]
+        [InlineData(
+            ExcludeWithWildCards,
+            @"**\*",
+            @"**\b\**",
+            new[] { "1.cs", @"a\2.cs", @"a\b\3.cs", @"a\b\c\4.cs" },
+            new[] { "1.cs", @"a\2.cs", "build.proj" })]
+        [InlineData(ExcludeWithWildCards,
+            @"**\*",
+            @"**\b\**\*.cs",
+            new[] { "1.cs", @"a\2.cs", @"a\b\3.cs", @"a\b\c\4.cs", @"a\b\c\5.txt" },
+            new[] { "1.cs", @"a\2.cs", @"a\b\c\5.txt", "build.proj" })]
+        [InlineData(
+            ExcludeWithWildCards,
+            @"src\**\proj\**\*.cs",
+            @"src\**\proj\**\none\**\*",
+            new[]
+            {
+                "1.cs",
+                @"src\2.cs",
+                @"src\a\3.cs",
+                @"src\proj\4.cs",
+                @"src\proj\a\5.cs",
+                @"src\a\proj\6.cs",
+                @"src\a\proj\a\7.cs",
+                @"src\proj\none\8.cs",
+                @"src\proj\a\none\9.cs",
+                @"src\proj\a\none\a\10.cs",
+                @"src\a\proj\a\none\11.cs",
+                @"src\a\proj\a\none\a\12.cs"
+            },
+            new[]
+            {
+                @"src\a\proj\6.cs",
+                @"src\a\proj\a\7.cs",
+                @"src\proj\4.cs",
+                @"src\proj\a\5.cs"
+            })]
+        [InlineData(
+            ExcludeWithWildCards,
+            @"**\*",
+            "foo",
+            new[]
+            {
+                "foo",
+                @"a\foo",
+                @"a\a\foo",
+                @"a\b\foo",
+            },
+            new string[]
+            {
+                @"a\a\foo",
+                @"a\b\foo",
+                @"a\foo",
+                "build.proj"
+            })]
+        public void ExcludeVectorWithWildCards(string projectContents, string includeString, string excludeString, string[] inputFiles, string[] expectedInclude)
+        {
+            //todo run the test twice with slashes and back-slashes :)
```

Comment:
Oh--that's actually interesting, since there are both filesystem and MSBuild-include-syntax impacts to this. I guess you can ignore my comment as long as creating files is doable with these strings.